pkgs/by-name/ob/obj2voxel/package.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "obj2voxel";
+  version = "1.3.4";
+
+  src = (
+    fetchFromGitHub {
+      owner = "Eisenwave";
+      repo = finalAttrs.pname;
+      rev = "refs/tags/v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-FkGQjYm15le1Na+2U1TgK4nNOZNs5bME5J87pjVFpDs=";
+    }
+  ).overrideAttrs (
+    _: {
+      GIT_CONFIG_COUNT = 1;
+      GIT_CONFIG_KEY_0 = "url.https://github.com/.insteadOf";
+      GIT_CONFIG_VALUE_0 = "git@github.com:";
+    }
+  );
+
+  patches = [
+    # pull missing definitions (https://github.com/Eisenwave/voxel-io/pull/3)
+    (
+      fetchpatch {
+        name = "add-std-includes.patch";
+        url = "https://web.archive.org/web/20230914180442/https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/3.diff";
```

```suggestion
        url = "https://github.com/Eisenwave/voxel-io/compare/d902568de2d4afda254e533a5ee9e4ad5fe7d2be...2f800f6aca8acd0cc18f9625b411c6dc2956f2ea.patch";
```

And likewise for the other patch?

Kind of excessive going through archive.org for these just to get a permalink, isn't it?

---

Changed that as requested. Still, I'd like to point out that @dotlambda previously asked for stable URLs from which building in the far future is possible (see https://github.com/NixOS/nixpkgs/pull/254299#discussion_r1323164906). The most straightforward way I know to achieve this is `archive.org`. `github.com` is known to lose commit information through garbage collection, e.g. in case of forced pushes.